connectors/class-connector-wordpress-seo.php

```diff
@@ -420,7 +420,7 @@ private function meta( $object_id, $meta_key, $meta_value ) {
 				/* translators: %1$s: a meta field title, %2$s: a post title, %3$s: a post type (e.g. "Description", "Hello World", "Post") */
 				__( 'Updated "%1$s" of "%2$s" %3$s', 'stream' ),
 				$field['title'],
-				$post->post_title,
+				str_replace( '%', '%%', $post->post_title ),
```

Comment:
Thank you for looking into this again. You found it out quicker, than I could explain it.

Yes, we should do this. In fact, we should go through _all_ the [connectors](https://github.com/xwp/stream/tree/develop/connectors), since I found a lot of them not escaping the values. See my [other issue](https://github.com/xwp/stream/pull/1486#pullrequestreview-2123401562), which stems from the Gravity Forms connector, not Yoast SEO.

Reply:
The usage of `(string) vsprintf( $message, $args )` replacement logic in the main `log()` appears to be very hidden and in most of the instances the `$args` are actually set to be stored in `$stream_meta` instead of being the replacement values. Sounds like this requires a larger change to make the behaviour consistent.

Some examples:

- [ACF connector](https://github.com/xwp/stream/blob/cf03e514b30fbed10d126ef4c4c255b25110aaa1/connectors/class-connector-acf.php#L244) using args for the actual replacement values.
- [Gravity Forms connector](https://github.com/xwp/stream/blob/cf03e514b30fbed10d126ef4c4c255b25110aaa1/connectors/class-connector-gravityforms.php#L823) using its own `sprintf()` instead of relying on argument forwarding to `vsprintf`.

Reply:
Yes, there is no real consistency in these connectors. But preventing fatal errors seems still necessary with some hot fixes, so that users won't get fatal errors.